### DISCUSSION TITLE
Tailwind use too many file handles on Windows (EMFile error)

### DISCUSSION THREAD

**Discussion by louis-bompart** [tailwindlabs/tailwindcss#11890] [Help]
Hey there, while updating TailwindCSS, I notice that since #11548, the build process of a large project I work on ([coveo/ui-kit](https://github.com/coveo/ui-kit)) has been failing on Windows due to EMFile errors.

I suspect that the # of handles available on Windows is lower than on other OSes.

For now, I reverted the changes through a `patch-package`in our repo.
I wanted to bring this problem to your attention, as you may want to introduce some kind of retry for EMFile errors (e.g. that is what graceful-fs does).

**Comment by CharlesPipkinimh** [comment]
Hy there i can see your post and i must say

Close Unnecessary Applications: If you're running multiple applications or processes concurrently, they might be using up available file handles. Close any unnecessary applications to free up resources.

Limit Concurrent Processes: If you're using a build tool that spawns multiple processes concurrently (e.g., for compiling CSS), you might be hitting the file handle limit. Consider limiting the number of concurrent processes or using tools that manage resources more efficiently. [https://www.adpworkforce-now.com/](http://adpworkforce-now.com/)
Use NPM Scripts: If you're using NPM to manage your project, consider creating NPM scripts for your development processes. NPM can manage the processes and resource limits more effectively.

Upgrade Node.js: Outdated versions of Node.js might have issues that have been resolved in newer releases. Make sure you're using the latest stable version of Node.js

Thanks and regards
CharlesPipkinimh

**Comment by louis-bompart** [comment]
> Hy there i can see your post and i must say
> 
> Close Unnecessary Applications: If you're running multiple applications or processes concurrently, they might be using up available file handles. Close any unnecessary applications to free up resources.
> 
> Limit Concurrent Processes: If you're using a build tool that spawns multiple processes concurrently (e.g., for compiling CSS), you might be hitting the file handle limit. Consider limiting the number of concurrent processes or using tools that manage resources more efficiently. [https://www.adpworkforce-now.com/](http://adpworkforce-now.com/) Use NPM Scripts: If you're using NPM to manage your project, consider creating NPM scripts for your development processes. NPM can manage the processes and resource limits more effectively.
> 
> Upgrade Node.js: Outdated versions of Node.js might have issues that have been resolved in newer releases. Make sure you're using the latest stable version of Node.js
> 
> Thanks and regards CharlesPipkinimh

(While you're profile, its age, and actions make me believe you are a bot, I will answer those suggestions...)

While the proposed measures are good stop-gap for the consumer of Tailwind, I think Tailwind should do like other build tools such as webpack and be conscious of the possible limitation of the filesystem. This is why webpack use fs-graceful
